### ISSUE TITLE
Destroyer - Whirlwind

### ISSUE THREAD

**Issue by Dwarf-LordPangolin** [tsunamayo/Skywanderers-Pathfinder-Group-Build#5]
Just wanted to let you know the rebuild of that 100m ship was done!
![20200126234108_1](https://user-images.githubusercontent.com/33041786/73152405-69f21900-4095-11ea-9bd5-fefd43617d80.jpg)

Converting the old one to a 3m ceiling was not possible, so I just built a new one over the weekend. This is a full 125m, so a "long destroyer." Like the others it has a quick and dirty placeholder interior; I'm just blocking out space for rooms with half-height walls and a straight hallway ATM, a less-boring interior will follow.

You've mentioned that crew will be a big element of the game (which is awesome!); since we don't know how much space they and their facilities will need, I'm just trying to budget as much space for them as possible. It also has a ton of free space inside for more ship systems once those are more fully implemented, so it should be future-proofed to later updates.

Until [this bug](https://github.com/tsunamayo/Skywanderers/issues/1577) is fixed, I've had to disconnect the turrets; taking control of the ship with them connected to the yoke drops FPS to 1 or less (ideally, once missiles are implemented most of the big turrets will be replaced with missile launchers anyway). Once that bug is fixed it'll be at 100%, but if you want to take a look at it now, here it is!

[Whirlwind Destroyer - 1st Testing Version.zip](https://github.com/tsunamayo/Skywanderers-Pathfinder-Group-Build/files/4115779/Whirlwind.Destroyer.-.1st.Testing.Version.zip)

**Comment by tsunamayo** [comment]
Okay amazing as always! It is great for motivation to see some great ship being made. I love the cockpit, with the upper window.
This faction is really taking shape. I keep you posted for the turret bug. 
Yes proceeding like you did with the interior is the way to go.
I think you can move on with the detailing! And add some landing gear as usual.
Cheers

**Comment by Dwarf-LordPangolin** [comment]
Just wanted to give you an update; I'm still plugging away on this. Sorry it's taking so long, there's a heck of a lot of space in it compared with the others. Hopefully be done in a few days! Till then, here's a few screenshots.

![20200214012150_1](https://user-images.githubusercontent.com/33041786/74510064-921cad00-4ec8-11ea-8ac9-096445a55474.jpg)

![20200214012209_1](https://user-images.githubusercontent.com/33041786/74510076-95b03400-4ec8-11ea-8e31-a3e82783d4b0.jpg)

![20200214012224_1](https://user-images.githubusercontent.com/33041786/74510083-9943bb00-4ec8-11ea-82a7-9793ab2f110e.jpg)

![20200214012238_1](https://user-images.githubusercontent.com/33041786/74510088-9b0d7e80-4ec8-11ea-81d1-fc0d381910d7.jpg)

**Comment by Dwarf-LordPangolin** [comment]
All finished!

![20200216171901_1](https://user-images.githubusercontent.com/33041786/74614826-0cc01500-50e1-11ea-8b86-4c72c719d1ee.jpg)

It should be good to go. The one thing that doesn't work ATM is the elevator; it won't retract all the way because of the clipping bug. However, based on what you posted in the bug reports, it should work fine once the next release goes live. :+1:

**Comment by tsunamayo** [comment]
Okay looking amazing! Thanks a lot for your help on this. Could you send the bp? Cheers!

**Comment by Dwarf-LordPangolin** [comment]
Wow, I feel stupid; thought I had!

[Whirlwind Destroyer - v2.zip](https://github.com/tsunamayo/Skywanderers-Pathfinder-Group-Build/files/4211370/Whirlwind.Destroyer.-.v2.zip)

**Comment by tsunamayo** [comment]
Hi @Dwarf-LordPangolin, same deal, I dont know if your ship was impacted by the latest change, and if you have a more recent version. Cheers!

**Comment by Dwarf-LordPangolin** [comment]
[Whirlwind Destroyer - v3.zip](https://github.com/tsunamayo/Skywanderers-Pathfinder-Group-Build/files/4532560/Whirlwind.Destroyer.-.v3.zip)

Okie dokes, this is the most recent version! Updated tons of stuff; among other things, it has the new starter block on it (starboard engineering room), so you don't have to plunk a new one down when exporting.

Also, I loved some of the changes Gray Shuko made in the loading screen art so much that I made a couple tweaks to make it resemble his drawing a bit more; no huge changes, just cosmetic stuff like the red stripe on the underside.

I am currently running into basically **all** of the bugs regarding turrets ATM, so it's hard to test how balanced it is in combat. In particular, they insist on clipping into the hull even when nothing is targeted. Other than that, things seem to be working smoothly on this end. Let me know what you think, and if anything needs changing! :)

**Comment by tsunamayo** [comment]
Amazing, will check this out asap! yeah congrats on making the covert art ;)
Okay what bugs are breaking the experience now (I know the hinge reset at 0, will fix asap). I need to fix all of that asap, I want a good combat experience. I already used that ship to optimize things a lot and fixed tons of bugs.
So I dont know if you updated the weapon load, I think we should have a cocktail with laser and beam. Also to be honest I think it was still lagging a bit, there is sooo much turret on that one.
For the time being until more optimization is done (and until we can say to mini turret to be in defense of small target only) I think we could remove the smaller one if that works for you.
I definitely want to spawn that ship as one of the big NPC.

**Comment by Dwarf-LordPangolin** [comment]
So ATM, the bugs affecting turrets on this (and others):
https://github.com/tsunamayo/Starship-EVO/issues/1844
https://github.com/tsunamayo/Starship-EVO/issues/1825
https://github.com/tsunamayo/Starship-EVO/issues/1816
And the one mentioned here, where they target friendly ships:
https://github.com/tsunamayo/Starship-EVO/issues/1840
I think that's all of them.

Sorry about the turret load; I definitely added too much redundancy there! I can remove the smaller ones, or I can reduce the total number of turrets by half; whichever you'd prefer.

I agree 100% on the weapon load cocktail thing, and I'm really pleased that it looks like the weapons in Starship EVO are going to have different strengths and weaknesses, instead of just being cosmetically different. I'm planning on making the small turrets use beams, because a light, fast-tracking turret with a hitscan weapon sounds ideal as an anti-fighter (maybe later anti-missile?) turret. Because the barrels don't attach yet to the small beams I was going to put that plan on hold until they do, but if you want I can implement it now.

The fixed, large gatling guns on my ships were originally meant to shoot the larger but still fast classes -- corvettes, frigates, etc. Since it looks like gatlings are going to have a very high spread, I could replace those with beams as well. In the case of the Whirlwind, that would give it small beam turrets, large gatling laser turrets, large beam guns, and large long-range laser guns.

**Comment by tsunamayo** [comment]
Okay I just added a feature where you specify the max / min target size for auto-fire. I think we can keep your small turret with a small target setting, so in big battle they wont fire too much - if that work liek that we will keep.
I am on the other bugs!
Let me check the ship and play with it a bit, and we will do a final pass on the weapon load to make it combat fit at this stage in the game.
Also no barrel for beam! They dont make much sense - I couldnt find an use for them. 
Cheers

**Comment by Dwarf-LordPangolin** [comment]
Makes sense! I will wait for your feedback on the weapon load, and put beams on the small turrets while waiting.

Edit: also, that option allowing us to specify target size for turret auto fire sounds _**awesome**_. It'll be a huge help in making sure that our turrets shoot at what they're meant to. :D

**Comment by tsunamayo** [comment]
yeah, as Sabre said it will also be important once we have munition. I just want to inverse ship category, it just feels too weird to have to input Min size = 5 and Max size = 0, I can feel people will get extra confused by that. I wanted to call the game Starship Zero, as 0 was the biggest cat...

**Comment by Dwarf-LordPangolin** [comment]
Actually, since we're starting to finalize the loadout, an important question: how do you want this ship to fight? Like, as a ship that the player can encounter in the game, what situations might they encounter it in, and what kind of threat/ally is it supposed to be? That's going to affect the weapons it has.

Right now, it's set up to act as an escort, shooting down smaller ships -- especially stuff like bombers -- before they can attack bigger, more expensive stuff like bases, freighters, or really big warships. That's why it has a lot of weak but rapid-fire guns, so it has a better chance of hitting small, fast targets. But if it's going to fighting ships of a similar mass to itself regularly -- or, in the case of Sabre's ship, nearly twice its mass -- it will need a different kind of loadout. That won't be a problem, but it would help to know what function it'll be serving.

**Comment by Dwarf-LordPangolin** [comment]
I've finally gotten around to updating the Whirlwind so that it has all the new features that the game has, but let me know if it's missing something. I also did another pass on it to further refine the appearance, and bring it in-line with the style I've been working on. In particular, I was never happy with the turrets as they were meant for the initial pass, so I finally sat down and worked on making a couple different types of them. Copy-paste was absolutely invaluable in this, and has already saved me hours of work, so thank you so much for that! I also cleaned up the logic in the command center, so it's not just plastered all over the wall anymore. The teleporter looks a bit out-of-place; if it could be moved on the .25m grid instead of the 1m grid, that would be helpful in blending it into designs more.

![20210102235740_1](https://user-images.githubusercontent.com/33041786/103472924-affd3580-4d58-11eb-9f87-c6932b94788d.jpg)

![20210103000529_1](https://user-images.githubusercontent.com/33041786/103472926-b55a8000-4d58-11eb-8644-fecab078998c.jpg)

![20210103000635_1](https://user-images.githubusercontent.com/33041786/103472928-b68bad00-4d58-11eb-9a93-8dcac358c36f.jpg)

![20210103001203_1](https://user-images.githubusercontent.com/33041786/103472929-b7bcda00-4d58-11eb-931b-705b5fede3a4.jpg)


One issue it has that I can't fix at the moment is that the navigation deck on top has two holes where the windows need to go; due to what I believe is [this bug](https://github.com/tsunamayo/Starship-EVO/issues/2646), I can't place the glass there ATM. Not a huge issue, but it will look a little funny.

[Whirlwind Destroyer - v5.zip](https://github.com/tsunamayo/Skywanderers-Pathfinder-Group-Build/files/5761276/Whirlwind.Destroyer.-.v5.zip)

**Comment by tsunamayo** [comment]
@Dwarf-LordPangolin Amazing, let me check this out asap! I will restart working on combat soon now that I have the system holomap to find pirate / enemy, so I will use this ship a lot. I didnt saw your previous reply, so for now it will need anti-light ship plus some decent weapon to fight ships of the same size. I do not plan to go really bigger than this for now.  But do not spend time on this yet, I need more testing at this point!
Cheers and happy new year!

**Comment by Dwarf-LordPangolin** [comment]
Hi Tsuna! You mentioned in your feedback on the Mistral's latest version that you'd like to see more military faction ships. I would _love_ to do a military faction for you. I actually already made a ship to potentially replace the Whirlwind about a year ago. Let me know if you like it. It'll probably need some small updates, like Void Pearl Storage, but it's 99% ready to go other than that.

![20241105155307_1](https://github.com/user-attachments/assets/850f26bb-c682-429c-8144-7e4407d0ef1a)
[[AAM] Firestorm Destroyer Mk.2.zip](https://github.com/user-attachments/files/17638444/AAM.Firestorm.Destroyer.Mk.2.zip)

It also probably needs fewer weapons, I kinda went overboard on it. ;)

Regarding a military faction, I'll do a short lore writeup of my idea for it, and you can tell me if it fits with your plans for the game. It won't be AAM; AAM is actually a shipyard company, I just tagged a bunch of stuff with AAM early on to show the manufacturer, before I knew how the factions/corporations would work.

**Comment by tsunamayo** [comment]
@Dwarf-LordPangolin Amazing, sorry I have had time to do a debrief yet but it is absolutely stellar! So much thing going on! Yeah maybe it is too much laser, but I plan to do an optimization pass on that soon so I would be using the ship to do that! Cheers

**Comment by tsunamayo** [comment]
@Dwarf-LordPangolin Okay so it is a true masterpiece! I absolutely love the style of this faction, it is so characteristic and just flow so well. I think it come down to a good balance between details and flat surface, painted vs black area, and just overall interesting silhouette.
 Better for me as a game asset, it reads extremely well at mid-long distance. I love how each outside details is integrated with the small decals on the side. 
The turrets are really simple and elegant. I keep you posted if I cant optimize those enough, as indeed there is tons of beams and would make the game lags.

The retractable landing pad is amazing. It is nice to have those airlock rooms. The corridors are so interesting with the two tones, and rounded lower volume and some pocket of details here and there. It is all so clean and neatly integrated.
The living quarter is lovely with the controllers and the plugged in PS34 haha. I like how the cantina has a windows on top. All the life supports are neatly integrated. 
The pilot instruments is so cleanly integrated, it reminds me of the look of 2001 A Space Odyssey.
There is so many utility rooms, it really feels like a capital ship. The "war room" was really cool, I was surprised by it, make it look quite menacing. Also special mention to the teleport room, "beam me up scotty". Your ladder design is really great as it is functional ^^

Please send me your lore, I will see how I can integrate this in the game, I will do a big push on factions.
